src/main/java/at/ac/univie/mminf/luceneSKOS/analysis/engine/jena/SKOSEngineImpl.java

```diff
@@ -227,24 +277,51 @@ public SKOSEngineImpl(InputStream inputStream, String format, List<String> langu
         searcher = new IndexSearcher(DirectoryReader.open(indexDir));
     }
 
+    private static String getHash(Reader reader) throws IOException, NoSuchAlgorithmException
```

Comment:
I suggest using an InputStream as param for this function. Then you can pass it into DigestInputStream to calculate the MessageDigest.